### ISSUE TITLE
no re-rendering of a visited document when tab-browsing

### ISSUE THREAD

**Issue by ralf3u** [xiota/geany-preview#27]
Actual behavior
When tab-browsing with shortcuts or with the mouse, for example on a html-document, which already have been visited in the Geany session, and with an image on top of the webpage, then I can see a re-rendering in the preview (see the code down below), although there was no change.

Expectation
When tab-browsing with shortcuts or with the mouse from one document to another document, the preview should do some kind of hide the preview or keep the preview in memory. So, when the user is tab-browsing back in the editor to the document, the preview should appear right away, so without re-rendering.

Remark
I think this would be a cool a feature. The content of the preview would appear then in light speed.

Test
This is the test-document (a picture named 1.jpg need to be added in the same level of the html-document):
```
<!DOCTYPE html>
<head>
<title>test</title>
</head>
<body>
<br><br><br><br><br>
<img src="1.jpg" style="width:300px">
<br><br><br><br><br>
<div>1agsjdghjhsgjhgaskdgkagdaksdgaksdgh</div>
<div>agsjdghjhsgjhgaskdgkag2daksdgaksdgh</div>
<div>agsjdghjhsgjhgaskdgkagd4aksdgaksdgh</div>
<div>agsjdghjhsgjhgaskdgkagda5ksdgaksdgh</div>
<div>agsjdghjhsgjhgaskdgkagdaksd6gaksdgh</div>
<div>agsjdghjhsgjhgaskdgka3gdaksdgaksdgh</div>
<body>
<html>
```
The settings in preview.conf are:
`snippet_html=false`
`extra_css=extra-media.css`

**Comment by xiota** [comment]
The update is b/c the document could have changed while the preview was hidden.    Rather than do a lot of bookkeeping to track changes, it's easier to just update.  There are also lots of times when the preview needs to be updated that are hard to detect, and it's worse to not update when needed than to update when not needed.

**Comment by ralf3u** [comment]
>The update is b/c the document could have changed while the preview was hidden. 

Yes, but then a bar automatically appears on top of the editor, when the tab is again in focus. So, there is no need to check if the document was changed while the preview was hidden. This is the test:
- open Geany and open a document, let's say an html-document
- open an other document, a txt-document (result: the txt-document in Geany is now in focus)
- open the same html-document in an other editor, like FeatherPad, change the code and save the document
- in Geany: tab-browse back to the html-document (result: a bar appears on top of the editor)

>Rather than do a lot of bookkeeping to track changes, it's easier to just update.

The test shows that a bookkeeping to track changes is not necessary.

>There are also lots of times when the preview needs to be updated that are hard to detect, ...

I don't know what that means. Is there an example?

@xiota By the way, I would like to understand how Preview works: When is an update executed? Does Preview gets a signal from the editor when there is a change in the editor? Or Preview is updating many times a second, so without getting signals from the editor? Thank you for an answer in advance.

**Comment by xiota** [comment]
> the WebKitGTK rendering engine is working faster

It's not faster.  The way content is loaded into the browser was changed.  New method is to update nodes, which is probably slower because it's in javascript.  But the page is more stable, which fixes some problems.
